sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnType.scala

```diff
@@ -493,7 +493,7 @@ private[columnar] trait DirectCopyColumnType[JvmType] extends ColumnType[JvmType
 }
 
 private[columnar] object STRING
-  extends NativeColumnType(PhysicalStringType(CollationFactory.DEFAULT_COLLATION_ID), 8)
+  extends NativeColumnType(PhysicalStringType(SqlApiConf.get.defaultStringType.collationId), 8)
```

Comment:
Do we need to change this? It looks this object does not care about collation

Reply:
@dbatomic I changed it, but am not sure if it should be default or just utf8_binary